core/state/statedb.go

```diff
@@ -372,12 +372,16 @@ func (self *StateDB) deleteStateObject(stateObject *StateObject) {
 // Retrieve a state object given my the address. Returns nil if not found.
 func (self *StateDB) GetStateObject(addr common.Address) (stateObject *StateObject) {
 	// Prefer 'live' objects.
+	self.lock.Lock()
```

Comment:
Ooops, sorry for the late reply I saw this while on the way to pick up family. Hmm, I'll have to think more about that. I haven't measured the overhead but I use the conventional wisdom of ReadLocks being cheaper than an all or nothing Mutex for reads. However, not a biggie, we can continue this offline :)